idd/Energy+.idd.in

```diff
@@ -64816,50 +65000,43 @@ WaterHeater:HeatPump,
        \object-list ZoneNames
        \note Used only if Inlet Air Configuration is ZoneAirOnly or ZoneAndOutdoorAir.
        \note Otherwise, leave field blank.
-  A14, \field Tank Object Type
```

Comment:
@nmerket @Myoldmopar I see two main concerns here.
1.  Uniqueness of names - It's not that we *need* to support the same name on two water heaters, but rather do we enforce unique names across all water heaters?  Currently, the only unique name enforcement that is universal is that all object names must be unique within a specific object-type.  There are places that enforce unique names across a group of similar objects (e.g. schedules).  I don't know about the waterheaters - would have to look at the code.

2. Consistency of input - The last time this was discussed (years ago), the development team consensus was to use the dual-field type+name approach when referencing other objects in HVAC system connections.  Clearly, we aren't consistent with this everywhere - we don't have object-type fields when referencing a schedule name.  At it's possible that a user could name a schedule the same as some other object.  EnergyPlus knows it's looking for a schedule.  The IDD \object-list tags are the only way to inform an interface that this field is a schedule name.
If we want to adopt a new "rule" for this, then let's have a discussion that includes the OpenStudio team and bounce it off other interface developers.  Maybe they don't care which way it's done, but I would expect that they want consistency.  @kbenne ?